### PR TITLE
Add reflex package

### PR DESCRIPTION
**Pull request by TabNahida**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/r/re-flex/xmake.lua [hunk 1]
```diff
@@ -32,10 +30,6 @@ package("re-flex")
                 add_files("src/*.cpp")
                 add_deps("re-flex")
         ]])
-        local configs = {}
-        if package:config("shared") then
-            configs.kind = "shared"
-        end
```
Comment (waruqi) [inline packages/r/re-flex/xmake.lua]: configs is nil

Comment (star-hengxing) [reply packages/r/re-flex/xmake.lua]: xmake will automatically add kind

```lua
    if configs.kind == nil then
        table.insert(configs, "--kind=" .. (package:config("shared") and "shared" or "static"))
    end
```

File: packages/r/re-flex/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    add_versions("v4.3.0", "1658c1be9fa95bf948a657d75d2cef0df81b614bc6052284935774d4d8551d95")
+
+    on_install(function (package)
+        io.writefile("xmake.lua",[[
+            add_rules("mode.debug", "mode.release")
+            set_languages("cxx11")
+            add_includedirs("include")
+            set_encodings("utf-8")
+            add_vectorexts("all")
```
Comment (star-hengxing) [inline packages/r/re-flex/xmake.lua:17]: cross ci failed, because gcc vectorext api passed `-march=native` flags @waruqi

Comment (waruqi) [reply packages/r/re-flex/xmake.lua:17]: What flags should `all` be mapped for cross platform?

File: packages/r/reflex/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package("reflex")
+    set_description("The reflex package")
+
+    add_urls("https://github.com/Genivia/RE-flex/archive/refs/tags/v$(version).tar.gz","https://github.com/Genivia/RE-flex.git")
+    add_versions("3.5.1", "e08ed24a6799a6976f6e32312be1ee059b4b6b55f8af3b433a3016d63250c0e4")
+    add_versions("4.3.0", "1658c1be9fa95bf948a657d75d2cef0df81b614bc6052284935774d4d8551d95")
+
+    add_includedirs("include")
```
Comment (waruqi) [inline packages/r/reflex/xmake.lua:8]: please remove it

File: packages/r/reflex/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                set_languages("cxx11")
+                add_includedirs("include")
+                add_files("src/*.cpp")
+                add_deps("reflexlib")
+            target_end()
+        ]])
+        local configs = {}
+        if package:config("shared") then
+            configs.kind = "shared"
+        end
```
Comment (waruqi) [inline packages/r/reflex/xmake.lua:39]: and here

**Comment by waruqi** [comment]
two same packages?

**Comment by TabNahida** [comment]
> two same packages? why?

Sorry. My other device have not correctly refresh the progress

File: packages/r/re-flex/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        io.writefile("xmake.lua",[[
+            add_rules("mode.debug", "mode.release")
+            set_languages("cxx11")
+            add_includedirs("include")
+            set_encodings("utf-8")
+
+            option("vectorexts")
+                set_default(false)
+                set_showmenu(true)
+                add_vectorexts("all")
```
Comment (waruqi) [inline packages/r/re-flex/xmake.lua:21]: this option is unused.

File: packages/r/re-flex/xmake.lua [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                if is_plat("windows") and is_kind("shared") then
+                    add_rules("utils.symbols.export_all", {export_classes = true})
+                end
+
+            target("reflex")
+                set_kind("binary")
+                add_files("src/*.cpp")
+                add_deps("re-flex")
+        ]])
+        import("package.tools.xmake").install(package, configs)
```
Comment (waruqi) [inline packages/r/re-flex/xmake.lua:38]: configs is nil

File: packages/r/re-flex/xmake.lua [hunk 7]
```diff
@@ -0,0 +1,49 @@
+package("re-flex")
+    set_homepage("https://www.genivia.com/doc/reflex/html")
+    set_description("A high-performance C++ regex library and lexical analyzer generator with Unicode support. Extends Flex++ with Unicode support, indent/dedent anchors, lazy quantifiers, functions for lex and syntax error reporting and more. Seamlessly integrates with Bison and other parsers.")
```
Comment (waruqi) [inline packages/r/re-flex/xmake.lua:3]: too long description

File: packages/r/re-flex/xmake.lua (hunk 2 above)
Comment (waruqi) [reply packages/r/re-flex/xmake.lua]: https://github.com/xmake-io/xmake/pull/5114

**Pull request merged**